src/firebolt/async_db/cursor.py

```diff
@@ -419,9 +419,16 @@ async def _do_execute(
                 if isinstance(query, SetParameter):
                     await self._validate_set_parameter(query)
                 elif async_execution:
-                    resp = await self._async_execution_api_request(query)
-                    await self._raise_if_error(resp)
-                    self._query_id = self._query_id_from_response_async(resp)
+                    response = await self._async_execution_api_request(query)
+                    await self._raise_if_error(response)
+                    if response.headers.get("content-length", "") == "0":
```

Comment:
Why did you have to inline this part?

Reply:
I think that might have been cruft from an earlier design decision that was changed. I removed it.